background.js

```diff
@@ -1,3 +1,79 @@
+"use strict";
+
+function isEmptyObject (obj) {
+    return (Object.keys(obj).length === 0 && obj.constructor === Object);
+}
+
+// Clean AMP URLs
+function clean_amp(url) {
```

Comment:
There's no perfect way to tell that an arbitrary URL is an AMP article just by looking at the URL. The function provided would fail, for example, [here](https://mashable.com/article/snapchat-spectacles-3-release.amp).

I think the only way to do it is to pull the text of the article, which will contain something like `<link data-page-subject="true" href="https://mashable.com/article/snapchat-spectacles-3-release/" rel="canonical" />`, which contains the true URL.

Reply:
Yes, I wanted to discuss this with you. Prefetching the URL it's a _sure_ way to obtain the real address, but it's kind of kludge. I wanted to investigate this soon, so no clue if this could our only option, _ugh_. Do you happen to know a bit more about AMP?

Reply:
My best solution would be to get the top 100 amp-using websites, create custom rules for each of them, and prefetch the URL for anything that's not in our list.

Reply:
That's a sensible compromise, it makes sense. Probaly we can get away even with ~20 links, who knows. I'll keep that in mind. Everytime I encounter a different AMP url I update the test case file `test_urls.html`). Feel free to add your links, too!

Reply:
I wonder if we could teach a neural net to sanitise amp links.

Reply:
I was reading a bit how others have approached the task of "deAMPifying" links (f.e. [here](https://www.xda-developers.com/bypass-amp-links-deampify/)). I'd like to also investigate your suggestion to retrieve the canonical URL either way and test how fast (or not fast) that could be. Also, interesting to read that for a mobile User Agent the canonical link is not embedded in the HTML content

Reply:
You made very good points on this, let's work on the de-ampiefier in a separate PR